### PR TITLE
Update the test_subplot_outside_plotting_positioning test and increase the RMS tolerance

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

The test `test_subplot_outside_plotting_positioning` fails in the latest run of the "GMT Legacy Tests" (see https://github.com/GenericMappingTools/pygmt/actions/runs/4474241635/jobs/7862497307). 

It fails due to an upstream bug that affects GMT 6.3.0 only (see https://github.com/GenericMappingTools/gmt/issues/6100). 

In this PR, I remove `frame=True` from the test so that it works with both GMT 6.3 and 6.4.

I also increase the RMS tolerance to 4.0 so that the test passes for GMT master branch.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Now the test passes in the Tests (https://github.com/GenericMappingTools/pygmt/actions/runs/4497107567?pr=2454), the "GMT Dev Tests" (https://github.com/GenericMappingTools/pygmt/actions/runs/4497014198) and also the "GMT Legacy Tests" (https://github.com/GenericMappingTools/pygmt/actions/runs/4497107573).

The two failures in the "GMT Legacy Tests" are related to the doctests, which will be addressed in https://github.com/GenericMappingTools/pygmt/issues/2286, https://github.com/GenericMappingTools/pygmt/pull/2443 and https://github.com/GenericMappingTools/pygmt/pull/2450.

File: pygmt/tests/test_subplot.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def test_subplot_nrows_ncols_less_than_one_error():
             pass
 
 
-@pytest.mark.mpl_image_compare
+@pytest.mark.mpl_image_compare(tolerance=4.0)
```
Comment (weiji14) [inline pygmt/tests/test_subplot.py:105]: Add a link to this PR for context?

```suggestion
# Increase tolerance for compatibility with GMT 6.3 and 6.4, see
# https://github.com/GenericMappingTools/pygmt/pull/2454
@pytest.mark.mpl_image_compare(tolerance=4.0)
```

File: .github/workflows/ci_tests_legacy.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ name: GMT Legacy Tests
 on:
   # push:
   #   branches: [ main ]
-  # pull_request:
+  pull_request:
```
Comment (seisman) [inline .github/workflows/ci_tests_legacy.yaml:8]:
```suggestion
  # pull_request:
```

Comment (seisman) [inline .github/workflows/ci_tests_legacy.yaml:8]:
```suggestion
  # pull_request:
```

Comment (seisman) [inline .github/workflows/ci_tests_legacy.yaml:8]:
```suggestion
  # pull_request:
```

**Review by weiji14 — APPROVED** [review]

**Pull request merged**